doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,143 @@
+---
+title: "PARTITION BY"
+description: "Specifying the sort order of the data."
+aliases:
+  - /guides/partition-by/
+  - /sql/patterns/partition-by/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+A few types of Materialize collections are durably written to storage: [materialized views](/sql/create-materialized-view/), [tables](/sql/create-table), and [sources](/sql/create-source).
+
+Internally, Materialize stores these durable collections in an [LSM-tree](https://en.wikipedia.org/wiki/Log-structured_merge-tree)-like structure. Each collection is made up of a set of
+**runs** of data, each run is sorted and then split up into individual **parts**, and those parts are written to object storage and retrieved only when necessary to satisfy a query. Materialize will also periodically **compact** the data it stores to consolidate small parts into larger ones or discard deleted rows.
+
+Materialize lets you specify the ordering it will use to sort these runs of data internally. A well-chosen sort order can unlock optimizations like [filter pushdown](#filter-pushdown), which in turn can make queries and other operations more efficient.
+
+## Syntax
+
+The option `PARTITION BY <column list>` declares that a [materialized view](/sql/create-materialized-view/#with_options), [table](/sql/create-table/#with_options), or source table should be ordered by the listed columns. For example, a table that stores an append-only collection of events may look like:
```

Comment:
I've updated the MV docs as part of this PR as well.

I have not updated the table docs yet: those do not have a separate section for "with options" yet. I'm happy to add such a section and move the retain-history option to it if that sounds good to you.

AFAICT "create table ... from source` is not documented anywhere yet, so I have nothing to link to here. I'm open to suggestions!

Reply:
> As for the create table page -- that does have a with_options section, yes?

Not in the same way as: https://materialize.com/docs/sql/create-materialized-view/#with_options

The MV page has a general syntax for with options, then a table for the options that's separate from the main syntax. The table page has only a single option right now I guess... it's just defined alongside the other syntax and without the same level of detail. Happy to expand it out to an MV-style doc if that feels like a good change, though.

Reply:
Oh ... yikes.  Yes, please.